### PR TITLE
chore: added secure logging

### PR DESCRIPTION
**Pull request by chejimmy**
# Description

A service log should be able to answer questions such as “who did what when?” without including sensitive information such as passwords or secrets.

# How Has This Been Tested?

sample log: `LOG [CognitoJwtAuthGuard] User ID "c443af37-2eab-4465-ac94-7ec8b814abad" authorized to DashboardsController.read (url: "/api/dashboards/M8Dsr97LK0fC")`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/core/src/auth/cognito-jwt-auth.guard.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ import { jwtConfig } from '../config/jwt.config';
 import { Reflector } from '@nestjs/core';
 import { isPublicMetadataKey } from './public.decorator';
 
+interface HttpRequest {
+  headers: { authorization: string | undefined } | undefined;
```
Comment (mnischay) [inline apps/core/src/auth/cognito-jwt-auth.guard.ts:15]: headers?: { authorization?: string };

File: apps/core/src/auth/cognito-jwt-auth.guard.ts [hunk 2]
```diff
@@ -52,8 +57,15 @@ export class CognitoJwtAuthGuard implements CanActivate {
     );
 
     try {
-      await this.config.cognitoJwtVerifier.verify(this.getBearerToken(context));
-      this.logger.log(`Request authorized to ${controllerName}.${handlerName}`);
+      const httpRequest = context.switchToHttp().getRequest<HttpRequest>(),
+        { url } = httpRequest;
```
Comment (mnischay) [inline apps/core/src/auth/cognito-jwt-auth.guard.ts:61]: nit: using seperate const is preferred

File: apps/core/src/auth/cognito-jwt-auth.guard.ts [hunk 3]
```diff
@@ -66,12 +78,9 @@ export class CognitoJwtAuthGuard implements CanActivate {
...
   }
 
-  private getBearerToken(context: ExecutionContext): string {
-    const { headers: { authorization = '' } = { authorization: '' } } = context
-      .switchToHttp()
-      .getRequest<{
-        headers: { authorization: string | undefined } | undefined;
-      }>();
+  private getBearerToken(httpRequest: HttpRequest): string {
+    const { headers: { authorization = '' } = { authorization: '' } } =
```
Comment (mnischay) [inline apps/core/src/auth/cognito-jwt-auth.guard.ts:82]: const { headers: { authorization } } = httpRequest;
return (authorization ?? '').replace(/^Bearer /, '');

might be a better way ?

**Review by mnischay — CHANGES_REQUESTED** [review]
juts minor nits

File: apps/core/src/auth/cognito-jwt-auth.guard.ts [hunk 4]
```diff
@@ -52,8 +57,15 @@ export class CognitoJwtAuthGuard implements CanActivate {
...
-      await this.config.cognitoJwtVerifier.verify(this.getBearerToken(context));
-      this.logger.log(`Request authorized to ${controllerName}.${handlerName}`);
+      const httpRequest = context.switchToHttp().getRequest<HttpRequest>(),
+        { url } = httpRequest;
+
+      const { sub } = await this.config.cognitoJwtVerifier.verify(
+        this.getBearerToken(httpRequest),
+      );
+      this.logger.log(
+        `User ID "${sub}" authorized to ${controllerName}.${handlerName} (url: "${url}")`,
```
Comment (dpportet) [inline apps/core/src/auth/cognito-jwt-auth.guard.ts:67]: Do we know if logging this token is okay with security?

Comment (chejimmy) [reply apps/core/src/auth/cognito-jwt-auth.guard.ts:67]: The `sub` is an UUID of the authenticated user, not any sensitive information (no way to carry out attack with `sub`) 👍

https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-id-token.html
> sub
A unique identifier (UUID), or subject, for the authenticated user. The username might not be unique in your user pool. The sub claim is the best way to identify a given user.

File: apps/core/src/auth/cognito-jwt-auth.guard.ts (hunk 3 above)
Comment (chejimmy) [reply apps/core/src/auth/cognito-jwt-auth.guard.ts]: in `const { headers: { authorization } } = httpRequest;`, the `headers` might be undefined so cannot do it that way.

updated to the following for simplicity:
```
    const authorization = httpRequest.headers?.authorization ?? '';

    return authorization.replace(/^Bearer /, '');
```

**Review by dpportet — APPROVED** [review]

**Pull request merged**